core/tx_pool.go

```diff
@@ -515,8 +515,8 @@ func (pool *TxPool) ContentFrom(addr common.InternalAddress) (types.Transactions
 // transactions and only return those whose **effective** tip is large enough in
 // the next pending execution environment.
 func (pool *TxPool) TxPoolPending(enforceTips bool, etxSet types.EtxSet) (map[common.AddressBytes]types.Transactions, error) {
-	pool.mu.Lock()
-	defer pool.mu.Unlock()
+	pool.mu.RLock()
+	defer pool.mu.RUnlock()
```

Comment:
Why did this change from Lock to RLock?

Reply:
More efficient

Reply:
I disagree. Lock() locks for reading and writing, while RLock() only locks for reading. RLock() is much faster and doesn't lock the txpool for writing.